doc/sphinx/language/gallina-specification-language.rst

```diff
@@ -77,13 +79,13 @@ Numerals
   integer.  Underscores embedded in the digits are ignored, for example
   ``1_000_000`` is the same as ``1000000``.
 
-  .. productionlist:: coq
-     numeral : `num`[. `num`][`exp`[`sign`]`num`]
-     int     : [-]`num`
-     num     : `digit`…`digit`
-     digit   : 0..9
-     exp     : e | E
-     sign    : + | -
+  .. insertprodn numeral digit
+
+  .. prodn::
+     numeral ::= {+ @digit } {? . {+ @digit } } {? {| e | E } {? {| + | - } } {+ @digit } }
```

Comment:
I think you're mistaken.  The `-` may be considered an operator rather than part of the numeral.

cLexer.ml has:

```
  | Some ('0'..'9') ->
      let n = NumTok.parse s in
      let ep = Stream.count s in
      comment_stop bp;
      (NUMERAL n, set_loc_pos loc bp ep)
```

and NumTok.parse doesn't look for a leading `-`.

`integer`, which gets renamed to `int` is defined as:

```
  integer:
    [ [ i = NUMERAL      -> { my_int_of_string loc (check_int loc i) }
      | "-"; i = NUMERAL -> { - my_int_of_string loc (check_int loc i) } ] ]
  ;
```

FWIW, the error message in `my_int_of_string` doesn't mention that decimal numbers are not permitted, either:

```
let my_int_of_string loc s =
  try
    int_of_string s
  with Failure _ ->
    CErrors.user_err ~loc (Pp.str "This number is too large.")
```

Reply:
OK, then do you think that the text should be changed? And can you open a PR to fix the `my_int_of_string` issue you noticed (or at least a new issue)?

Reply:
I take it back, the `check_int` covers that case.  So nothing to change.  But OCaml has this little gem :-(:

```
# 4611686018427387904 = -4611686018427387904;;
- : bool = true
```

Because 

```
# 4611686018427387904;;
- : int = -4611686018427387904
```